schemas/host-config.json

```diff
@@ -360,6 +360,16 @@
 				},
 				"errorMessage": {
 					"$ref": "#/definitions/ErrorMessageConfig"
+				},
+				"validationBehavior":{
+					"type":"string",
+					"description":"Controls when inputs are validated",
+					"default":"onFocusLost",
+					"enum":[
+						"onFocusLost",
```

Comment:
What about `onInputChange`? I can see hosts wanting validation to start as soon as a user starts typing, rather than when focus is lost.

Also worth considering `immediate`, where the host would start validation even before user interacts (e.g. initially empty required inputs, or a default that fails validation).

Reply:
We did have one respondent to our survey that chose "other" that requested basically the `onInputChange` behavior as you describe it, so it may be a reasonable option.

That said, based on reading up on this a bit using the research a previous PM did for us, I personally don't think that validating sooner than focus lost is a good idea. See for example Pitfall #1 at [https://baymard.com/blog/inline-form-validation](https://baymard.com/blog/inline-form-validation) for some discussion of this. I'm hesitant to give more options to hosts here that i think may make things worse for the user.

On the other hand, left to my own devices i'd probably remove this setting altogether and make everyone use onFocusLost, so it's possible my bias is unduly prescriptivist :)

---
In reply to: [472681978](https://github.com/microsoft/AdaptiveCards/pull/4603#discussion_r472681978) [](ancestors = 472681978)